### PR TITLE
Add a profiles testsuite for jpa-map storage

### LINKED ISSUE #12045: Add a profiles testsuite for jpa-map storage
Add maven profile to both `testsuite/model` and `testsuite/integration-arquillian/tests/base` pom.xml for testing jpa map storage implementation.

### PR DESCRIPTION
**Pull request by vramik**
Closes #12045

### CODE REVIEW

File: testsuite/model/src/test/java/org/keycloak/testsuite/model/ClientModelTest.java [hunk 1]
```diff
@@ -122,10 +122,8 @@ public void testClientsBasics() {
             });
 
             withRealm(realmId, (session, realm) -> {
-                ClientModel client = session.clients().searchClientsByClientIdStream(realm, "%", 0, 10).findFirst().orElse(null);
-                assertThat(client, notNullValue());
-                assertThat(client.getId(), is(equalTo(searchClient.getId())));
-                assertThat(client.getClientId(), is(equalTo(searchClientId)));
+                // when searching by "%" all entries are expected
+                assertThat(session.clients().searchClientsByClientIdStream(realm, "%", 0, 10).count(), is(equalTo(2L)));
```
Comment (sguilhen) [inline testsuite/model/src/test/java/org/keycloak/testsuite/model/ClientModelTest.java:126]: Just out of curiosity: is this change required for this PR?

Comment (vramik) [reply testsuite/model/src/test/java/org/keycloak/testsuite/model/ClientModelTest.java:126]: Without it model tests fails, because it's not guaranteed which of the two clients is returned. 

As we are skipping the profile in GHA it won't be visible there, but locally it fails. We can move it to another issue if it makes more sense.

Comment (sguilhen) [reply testsuite/model/src/test/java/org/keycloak/testsuite/model/ClientModelTest.java:126]: I thought you'd say that, just wanted to make sure that was really the case ;) LGTM

**Review by sguilhen — APPROVED** [review]

**Review by hmlnarik — APPROVED** [review]
Approving per @sguilhen 's approval.
